dot/core/consensus.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/dot/core/types"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+
+	log "github.com/ChainSafe/log15"
+)
+
+// newBabeSession creates a new BABE session
+func (s *Service) newBabeSession() (*babe.Session, error) {
```

Comment:
This method was code extracted from `handleBabeSession`, which has been updated below in `service.go`. This method and the other methods within this file is the bulk of new code within this pull request. I will comment where other items have simply moved from one file to another.

Reply:
I also think this method in particular could probably be refactored at a later point and included in babe as a reinitialization method that gets passed a couple params if need be.